cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -386,50 +413,57 @@ synchronized UserTaskInfo getUserTaskByUserTaskId(UUID userTaskId, HttpServletRe
           new UserTaskInfo(httpServletRequest, new ArrayList<>(Collections.singleton(operation.apply(userTaskId.toString()))),
                            _time.milliseconds(), userTaskId, TaskState.ACTIVE);
-      _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).put(userTaskId, userTaskInfo);
+      _uuidToActiveUserTaskInfoMap.put(userTaskId, userTaskInfo);
     }
-    return _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).get(userTaskId);
+    return _uuidToActiveUserTaskInfoMap.get(userTaskId);
   }
 
-  public synchronized List<UserTaskInfo> getActiveUserTasks() {
-    return new ArrayList<>(_allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).values());
-  }
-
-  public synchronized List<UserTaskInfo> getCompletedUserTasks() {
-    return new ArrayList<>(_allUuidToUserTaskInfoMap.get(TaskState.COMPLETED).values());
+  public synchronized List<UserTaskInfo> getAllUserTasks() {
+    List<UserTaskInfo> allUserTasks = new ArrayList<>(_uuidToActiveUserTaskInfoMap.values());
+    if (_inExecutionUserTaskInfo != null) {
+      allUserTasks.add(_inExecutionUserTaskInfo);
+    }
+    allUserTasks.addAll(_uuidToCompletedUserTaskInfoMap.values().stream().flatMap(map -> map.values().stream())
+                                                       .collect(Collectors.toList()));
+    return allUserTasks;
   }
 
   @Override
   public String toString() {
-    Map<UUID, UserTaskInfo> completedWithSuccessUserTaskIdToFuturesMap = new LinkedHashMap<>();
-    Map<UUID, UserTaskInfo> completedWithErrorUserTaskIdToFuturesMap = new LinkedHashMap<>();
-    _allUuidToUserTaskInfoMap.get(TaskState.COMPLETED).forEach((k, v) -> {
-      if (v.state() == TaskState.COMPLETED) {
-        completedWithSuccessUserTaskIdToFuturesMap.put(k, v);
-      } else {
-        completedWithErrorUserTaskIdToFuturesMap.put(k, v);
-      }
-    });
+    Map<UUID, UserTaskInfo> uuidToCompletedWithSuccessUserTaskInfoMap = new LinkedHashMap<>();
+    Map<UUID, UserTaskInfo> uuidToCompletedWithErrorUserTaskInfoMap = new LinkedHashMap<>();
+    for (Map<UUID, UserTaskInfo> infoMap : _uuidToCompletedUserTaskInfoMap.values()) {
+      infoMap.forEach((k, v) -> {
+        if (v.state() == TaskState.COMPLETED) {
+          uuidToCompletedWithSuccessUserTaskInfoMap.put(k, v);
+        } else {
+          uuidToCompletedWithErrorUserTaskInfoMap.put(k, v);
+        }
+      });
+    }
     return "UserTaskManager{_sessionKeyToUserTaskIdMap=" + _sessionKeyToUserTaskIdMap
-           + ", _activeUserTaskIdToFuturesMap=" + _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE)
-           + ", _completedWithSuccessUserTaskIdToFuturesMap=" + completedWithSuccessUserTaskIdToFuturesMap
-           + ", _completedWithErrorUserTaskIdToFuturesMap=" + completedWithErrorUserTaskIdToFuturesMap + '}';
+           + ", _uuidToActiveUserTaskInfoMap=" + _uuidToActiveUserTaskInfoMap
+           + ", _inExecutionUserTask=" + (_inExecutionUserTaskInfo != null ? _inExecutionUserTaskInfo : "None")
```

Comment:
Can we make the following change to cover anomaly detector started executions:
`None` -> `No-User-Initiated-Execution`